omero/sysadmins/unix/install-web/walkthrough/omeroweb-install-debian10-ice3.6.rst

```diff
@@ -0,0 +1,259 @@
+
+Install `WhiteNoise <http://whitenoise.evans.io/>`_::
+
+    /opt/omero/web/venv3/bin/pip install --upgrade 'whitenoise<4'
+
+
+**The following steps are run as the omero-web system user.**
+
+
+Configure WhiteNoise and start OMERO.web manually to test the installation::
+
+    omero config append -- omero.web.middleware '{"index": 0, "class": "whitenoise.middleware.WhiteNoiseMiddleware"}'
+
+    omero web start
+
+    # Test installation e.g. curl -sL localhost:4080
+
+    omero web stop
+
+
+Automatically running OMERO.web
+-------------------------------
+
+
+**The following steps are run as root.**
+
+Should you wish to run OMERO.web automatically, a `init.d` file could be created. See below an example file `omero-web-init.d`::
+
+    #!/bin/bash
+    #
+    # /etc/init.d/omero-web
+    # Subsystem file for "omero" web
+    #
+    ### BEGIN INIT INFO
+    # Provides:             omero-web
+    # Required-Start:       $local_fs $remote_fs $network $time omero postgresql
+    # Required-Stop:        $local_fs $remote_fs $network $time omero postgresql
+    # Default-Start:        2 3 4 5
+    # Default-Stop:         0 1 6
+    # Short-Description:    OMERO.web
+    ### END INIT INFO
+    #
+    ### Redhat
+    # chkconfig: - 98 02
+    # description: init file for OMERO.web
+    ###
+
+    RETVAL=0
+    prog=omero-web
+
+    # Read configuration variable file if it is present
+    [ -r /etc/default/$prog ] && . /etc/default/$prog
+
+    OMERO_USER=${OMERO_USER:-omero-web}
+    OMERO=/opt/omero/web/venv3/bin/omero
+    OMERODIR=/opt/omero/web/omero-web
+    VENVDIR=${VENVDIR:-/opt/omero/web/venv3}
+
+    start() {
+        echo -n $"Starting $prog:"
+        su - ${OMERO_USER} -c ". ${VENVDIR}/bin/activate;OMERODIR=${OMERODIR} ${OMERO} web start" &> /dev/null && echo -n ' OMERO.web'
+        sleep 5
+        RETVAL=$?
+        [ "$RETVAL" = 0 ]
+            echo
+    }
+
+    stop() {
+        echo -n $"Stopping $prog:"
+        su - ${OMERO_USER} -c ". ${VENVDIR}/bin/activate;OMERODIR=${OMERODIR} ${OMERO} web stop" &> /dev/null && echo -n ' OMERO.web'
+        RETVAL=$?
+        [ "$RETVAL" = 0 ]
+            echo
+    }
+
+    status() {
+        echo -n $"Status $prog:"
+        su - ${OMERO_USER} -c ". ${VENVDIR}/bin/activate;OMERODIR=${OMERODIR} ${OMERO} web status"
+        RETVAL=$?
+    }
+
+    case "$1" in
+        start)
+            start
+            ;;
+        stop)
+            stop
+            ;;
+        restart)
+            stop
+            start
+            ;;
+        status)
+            status
+            ;;
+        *)
+            echo $"Usage: $0 {start|stop|restart|status}"
+            RETVAL=1
+    esac
+    exit $RETVAL
+
+Copy the `init.d` file, then configure the service::
+
+    cp omero-web-init.d /etc/init.d/omero-web
+    chmod a+x /etc/init.d/omero-web
+
+    update-rc.d -f omero-web remove
+    update-rc.d -f omero-web defaults 98 02
+
+
+
+Start up services::
+
+
+
+    cron
```

Comment:
Interesting, I didn't expect that. Is it not running?

Reply:
resolved in omeroweb-install